test/services/return-requirements/delete-note.service.test.js

```diff
@@ -0,0 +1,64 @@
+// write test in hapi/yar for delete-note.service.js
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const DatabaseSupport = require('../../support/database.js')
+const SessionHelper = require('../../support/helpers/session.helper.js')
+
+// Thing under test
+const DeleteNoteService = require('../../../app/services/return-requirements/delete-note.service.js')
+
+const sessionData = {
+  data: {
+    id: 'f1288f6c-8503-4dc1-b114-75c408a14bd0',
+    checkYourAnswersVisited: true,
+    licence: {
+      endDate: null,
+      licenceRef: '01/ABC',
+      licenceHolder: 'Astro Boy',
+      currentVersionStartDate: '2023-02-08T00:00:00.000Z'
+    },
+    reason: 'abstraction-below-100-cubic-metres-per-day',
+    journey: 'no-returns-required',
+    note: {
+      content: 'Note attached to requirement',
+      userEmail: 'carol.shaw@atari.com'
+    },
+    startDateOptions: 'licenceStartDate'
+  }
+}
+describe('Delete Note service', () => {
+  let session
+  let yarStub
+
+  beforeEach(async () => {
+    await DatabaseSupport.clean()
+
+    session = await SessionHelper.add({
+      ...sessionData
+    })
+
+    yarStub = {
+      flash: Sinon.stub()
+    }
+  })
+
+  it('deletes the note', async () => {
+    await DeleteNoteService.go(session.id, yarStub)
+    expect(session.note).to.be.undefined()
```

Comment:
First, a NIT, adding some whitespace. But also this test is incorrect.

It is currently passing because we never set the property `session.note` but _do_ set the property `session.data.note`. If you update the test to check for that it will fail because `session.data.note` is defined.

You need to do this to fix the test and properly check the note has been deleted.

```suggestion
    await DeleteNoteService.go(session.id, yarStub)
    
    const refreshedSession = await session.$query()
    
    expect(refreshedSession.data.note).to.be.undefined()
```

However, this still won't pass unless the `DeleteNoteService` is updated to drop the `note` property instead of setting it to an empty string.

Reply:
makes sense, updated